src/vm/functions/define.rs

```diff
@@ -53,7 +53,8 @@ pub enum DefineResult {
 fn check_legal_define(name: &str, contract_context: &ContractContext) -> Result<()> {
     use vm::is_reserved;
 
-    if is_reserved(name) || contract_context.variables.contains_key(name) || contract_context.functions.contains_key(name) {
+    if is_reserved(name) || contract_context.variables.contains_key(name) || contract_context.functions.contains_key(name) ||
```

Comment:
Maybe make this check a function of its own?  Looks unruly.

Reply:
Moved into ContractContext in 37ac38cd2